pkg/sbproxy/sbproxy.go

```diff
@@ -0,0 +1,234 @@
+}
+
+// Use provides a way to plugin middleware in the SBProxy
+func (s SBProxy) Use(middleware func(handler http.Handler) http.Handler) {
+	s.Server.Router.Use(middleware)
+}
+
+// Run is the entrypoint of the SBProxy. Run boots the application.
+func (s SBProxy) Run() {
+	var err error
+	ctx, cancel := context.WithCancel(context.Background())
+	defer waitWithTimeout(&group, time.Duration(s.AppConfig.TimeoutSec)*time.Second)
+	defer cancel()
+
+	handleInterrupts(ctx, cancel)
+
+	s.CronScheduler.Start()
+	defer s.CronScheduler.Stop()
+
+	addr := ":" + strconv.Itoa(s.AppConfig.Port)
+
+	logrus.Info("Running SBProxy server...")
+	if s.AppConfig.TLSKey != "" && s.AppConfig.TLSCert != "" {
+		err = s.Server.RunTLS(ctx, addr, s.AppConfig.TLSCert, s.AppConfig.TLSKey)
+	} else {
+		err = s.Server.Run(ctx, addr)
+	}
+	if err != nil && err != context.Canceled && err != context.DeadlineExceeded {
+		logrus.WithError(errors.WithStack(err)).Errorln("Error occurred while sbproxy was running")
+	}
+}
+
+// AddJob provides a way to add additional cron jobs to run alongside the SBProxy application.
+func (s *SBProxy) AddJob(schedule string, job cron.Job) error {
+	return s.CronScheduler.AddJob(schedule, job)
+}
+
+func defaultOSBServer(config *osb.ClientConfiguration) (*osbserver.Server, error) {
+	businessLogic, err := osb.NewBusinessLogic(config)
+	if err != nil {
+		return nil, err
+	}
+
+	reg := prom.NewRegistry()
+	osbMetrics := metrics.New()
+	reg.MustRegister(osbMetrics)
+
+	api, err := rest.NewAPISurface(businessLogic, osbMetrics)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating OSB API surface")
+	}
+
+	osbServer := osbserver.New(api, reg)
+	router := mux.NewRouter()
+
+	err = registerRoutes(Path, osbServer.Router, router)
+	if err != nil {
+		return nil, err
+	}
+
+	osbServer.Router = router
+	return osbServer, nil
+}
+
+func registerRoutes(prefix string, fromRouter *mux.Router, toRouter *mux.Router) error {
+	subRouter := toRouter.PathPrefix(prefix).Subrouter()
+	return fromRouter.Walk(func(route *mux.Route, router *mux.Router, ancestors []*mux.Route) error {
+
+		path, err := route.GetPathTemplate()
+		if err != nil {
+			return errors.Wrap(err, "error getting path template")
+		}
+
+		methods, err := route.GetMethods()
+		if err != nil {
+			return errors.Wrap(err, "error getting route methods")
+		}
+		logrus.Info("Registering route with methods: ", methods, " and path: ", path)
+		subRouter.Handle(path, route.GetHandler()).Methods(methods...)
+		return nil
+	})
+}
+
+func defaultRegJob(group *sync.WaitGroup, platformClient platform.Client, smConfig *sm.ClientConfiguration, proxyHost string) (cron.Job, error) {
+	smClient, err := smConfig.CreateFunc(smConfig)
+	if err != nil {
+		return nil, err
+	}
+	regTask := NewTask(group, platformClient, smClient, proxyHost)
+
+	return regTask, nil
+}
+
+//TODO: should happen earlier (ideally in sbproxy init(), logger.DefaultConfig()?)
+func setUpLogging(logLevel string, logFormat string) {
```

Comment:
maybe change the output to std.out as the default is err and looks weird in the logs